### DISCUSSION TITLE
Deselect the node when pressing `alt` and click

### DISCUSSION THREAD

**Discussion by Max-ChenFei** [xyflow/xyflow#2879] [Q&A]
Hi, here I change the `selected` pros, but the reactflow still does not work.
```
onNodeClick={(event, node) => {
        if (event.altKey) {
          console.log('alt click');
          console.log(node.selected);
          node.selected = !node.selected;
          node.label = 'test';
          console.log(node.selected);
        }
```

Thanks for your help.

**Comment by bcakmakoglu** [comment]
Nodes and edges aren't mutable refs or something, so doing `node.selected = !node.selected` will do nothing.

You need to use `setNodes`.

```ts
setNodes(nodes => nodes.map(node => ({ ...n, selected: !n.selected })))
```

**Reply by Max-ChenFei** [reply]
Thanks for your help. I want to toggle the selection of the node, but still it does not work. Maybe because of the conflicts with the default selecton.

**Reply by Max-ChenFei** [reply]
Hi, if want to toggle the selection of the node when pressing the `Alt`, 
```
 onNodeClick={(event, n) => {
        if (event.altKey) {
          setNodes((nds) =>
            nds.map((node) => {
              if (node.id === n.id) {
                node = {
                  ...n,
                  selected: !n.selected,
                };
              }
              return node;
            })
          );
        }
      }}
```
I see the style class changed when clicking, but the react component seems not to update.
![Animation](https://user-images.githubusercontent.com/28513798/222900443-07537e60-1c3e-42f7-afeb-db3e86fa8099.gif)
